src/email.rs

```diff
@@ -27,34 +37,39 @@ impl Config {
     ///
     /// If the attachment MIME type cannot be determined, it will default to
     /// `application/octet-stream`.
-    pub fn send_email(&self, email_type: Email) -> Result<()> {
+    pub fn send_email(&self, email_type: Email) -> Result<(), EmailError> {
         let email = self.create_email(email_type)?;
 
         // SMTP client setup
         let creds = Credentials::new(self.username.clone(), self.password.clone());
-        let tls = TlsParameters::new_rustls(self.smtp_server.clone())?;
-        let mailer = SmtpTransport::relay(&self.smtp_server)?
+        let tls = TlsParameters::new_rustls(self.smtp_server.clone())
+            .map_err(|_| EmailError::SendError)?;
+        let mailer = SmtpTransport::relay(&self.smtp_server)
+            .map_err(|_| EmailError::SendError)?
             .port(self.smtp_port)
             .credentials(creds)
             .tls(Tls::Required(tls))
             .build();
 
-        // Send the email
-        mailer.send(&email)?;
-
-        Ok(())
+        //Tries to Send the email
+        match mailer.send(&email) {
+            Ok(_) => Ok(()),
+            Err(_) => return Err(EmailError::SendError),
+        }
```

Comment:
Also returns a [lettre::transport::smtp::Error](https://docs.rs/lettre/latest/lettre/transport/smtp/struct.Error.html) please use that one.